sphinx/application.py

```diff
@@ -1215,9 +1219,9 @@ def add_search_language(self, cls: Any) -> None:
         .. versionadded:: 1.1
         """
         logger.debug('[app] adding search language: %r', cls)
-        from sphinx.search import SearchLanguage, languages
-        assert issubclass(cls, SearchLanguage)
-        languages[cls.lang] = cls
+        from sphinx.search import languages
+        lang = cast(str, cls.lang)  # SearchLanguage subclasses must set 'lang' to a str
```

Comment:
I had thought that ``cast`` is meant to be avoided?
```suggestion
        lang = cls.lang
        assert isinstance(lang, str)  # SearchLanguage subclasses must set 'lang' to a str
```